src/analyze/analyze-time-data.c

```diff
@@ -254,14 +287,15 @@ int acquire_time_data(sd_bus *bus, bool require_finished, UnitTimes **out) {
                 return bus_log_parse_error(r);
 
         while ((r = bus_parse_unit_info(reply, &u)) > 0) {
-                UnitTimes *t;
+                _cleanup_(unit_times_donep) UnitTimes *t = NULL;
```

Comment:
This feels a little dirty because it violates the naming convention of these functions. `_donep` is generally used for owned things on the stack (i.e. `_cleanup_` is not applied to a pointer), but this is a borrowed reference to heap memory. The other naming convention is `_freep`, which is also definitely wrong for this case, because we do not want to free *the pointer*, just deinitialize its pointee.

So it is kind of weird that this is a `DEFINE_TRIVIAL_CLEANUP_FUNC` called `donep` that takes a pointer-to-pointer and returns a pointer.

I chose to scope the weird `DEFINE_TRIVIAL_CLEANUP_FUNC_FULL` to just be in this file to avoid code far away being affected, at least, but I am not the happiest about the naming here.